enhancements/network/persistent-ips-sdn-secondary-networks.md

```diff
@@ -0,0 +1,1012 @@
+`IPAMLease` allocation will not be deleted if the CNI DEL arguments
+feature any owner indication. Thus, on the VM scenario, the `IPAMLease`s
+will **not** be deleted.
+
+#### Starting a (previously stopped) Virtual Machine
+```mermaid
+sequenceDiagram
+
+actor user
+
+user ->> KubeVirt: startVM(vm=vmA)
+
+KubeVirt ->> Kubernetes: createPod(name=<generated name>, owner=vmA)
+
+
+note over OVN-K: notices a new pod was added having an owner
+OVN-K ->> Kubernetes: getPersistentIPs(owner=vmA)
+Kubernetes -->> OVN-K: persistentIPs
+OVN-K ->> Kubernetes: result(..., ips = persistentIPs)
+
+Kubernetes -->> KubeVirt: OK
+
+KubeVirt -->> user: OK
+```
+
+When creating the pod, the OVN-Kubernetes IPAM module finds existing
+`IPAMLease`s for the workload. It will thus use those already reserved
+allocations, instead of generating brand new allocations for the pod where the
+encapsulating object will run.
+
+#### Removing a Virtual Machine
+```mermaid
+sequenceDiagram
+
+actor user
+
+user ->> KubeVirt: stopVM(vm=vmA)
+
+KubeVirt ->> Kubernetes: deletePod(owner=vmA)
+
+note over OVN-K: notices a pod was deleted
+OVN-K ->> OVN-K: ips = teardownOVNEntities()
+
+Kubernetes -->> KubeVirt: OK
+
+KubeVirt -->> user: OK
+
+loop persistentIPs with owner=vmA:
+    Kubernetes ->> Kubernetes: garbageCollect(allocation)
+end
+```
+
+This scenario, from the OVN-Kubernetes IPAM module perspective is exactly like
+deleting a pod: since the `IPAMLease` is owned by a Kubernetes entity we
+give up on manipulating the datastore for it, and delegate to Kubernetes to
+garbage collect them once the Virtual Machine is removed.
+
+### API Extensions
+
+We plan on adding a new CRD that would be hosted in the k8snetworkplumbingwg,
+and hopefully accepted as part of the
+[Kubernetes multi-networking de-facto standard](https://github.com/k8snetworkplumbingwg/multi-net-spec).
+Meaning we envision other plugins with IPAM functionality to use this CRD to
+provide persistent IPs for their workloads.
+
+- Adds a new `IPAMLease` CRD. OVN-Kubernetes will provision these to
+  "block" the IP allocation during the VM object lifecycle.
+  - It must be "owned" by the pod's encapsulating object (i.e. VM) - thus
+  allowing Kubernetes to GC it once they're gone. To comply with Kubernetes GC,
+  the `IPAMLease` **must** be created in the same namespace as its owner.
+  - It must be "retrievable" by OVN-Kubernetes (i.e. so the "old" VM gets the
+  same IP when in runs in a "new" pod). For that, we will compose a unique key
+  with the required information, allowing the clients to get the CR by name.
+
+This is the proposed spec of the CRD:
+```yaml
+apiVersion: "persistentip.cni.cncf.io/v1alpha1"
+kind: IPAMLease
+metadata:
+  name: vm-a.tenantred.pod16367aacb67
+  namespace: ns1
+  ownerReferences:
+  - apiVersion: kubevirt.io/v1
+    kind: VirtualMachine
+    name: vm-a
+    uid: a0790345-4e84-4257-837a-e3d762d191ab
+spec:
+  network: tenantred
+  interface: pod16367aacb67
```

Comment:
Maybe I could add this info to an "upgrade" related section ? 

Never the less, the approach I'm focusing on now addresses it (since virt-controller instructs the CNI to look for an `IPAMClaim` named after the `VMI.Spec.Network` name, which does **not** change in the scenario you present.

Never the less, thanks for raising. Good catch.